hierarchicalforecast/core.py

```diff
@@ -49,27 +55,27 @@ def _reverse_engineer_sigmah(Y_hat_df, y_hat, model_name):
     direct usage of an estimated $\hat{sigma}_{h}$
     """
 
-    drop_cols = ['ds']
-    if 'y' in Y_hat_df.columns:
-        drop_cols.append('y')
+    drop_cols = [time_col]
+    if target_col in Y_hat_df.columns:
+        drop_cols.append(target_col)
     if model_name+'-median' in Y_hat_df.columns:
         drop_cols.append(model_name+'-median')
-    model_names = Y_hat_df.drop(columns=drop_cols, axis=1).columns.to_list()
+    model_names = Y_hat_df.drop(drop_cols).columns
```

Comment:
This will be very expensive for pandas dataframes, because a copy of the dataframe with the columns removed is made, only to extract the column names.
```suggestion
    model_names = [c for c in Y_hat_df.columns if c not in drop_cols]
```

Reply:
THanks, forgot that drop copies, good point